RELEASE_NOTES.md

```diff
@@ -1,5 +1,521 @@
 This file holds "in progress" release notes for the current release under development and is intended for consumption by the Chef Documentation team. Please see <https://docs.chef.io/release_notes/> for the official Chef release notes.
 
+# Chef Infra Client 16
+
+## Breaking Changes
+
+### Log Resource Notification Behavior
+
+The log resource in a recipe or resource will no longer trigger notifications by default. This allows authors to more liberally use log resources without impacting the updated resources count or impacting reporting to Automate. This change will impact users that used the log resource to aggregate notifications from other resources so they could limit the number of times a notification would fire. If you used the log resource to aggregate multiple notifications, you should convert to using the notify group resource which was introduced in Chef Infra Client 15.8.
+
+Example of notification aggregation with log resource:
+
+```ruby
+template '/etc/foo' do
+  source 'foo.erb'
+  notifies :write, 'log[Aggregate notifications using a single log resource]', :immediately
+end
+
+template '/etc/bar' do
+  source 'bar.erb'
+  notifies :write, 'log[Aggregate notifications using a single log resource]', :immediately
+end
+
+log 'Aggregate notifications using a single log resource' do
+  notifies :restart, 'service[foo]', :delayed
+end
+```
+
+Example of notification aggregation with notify_group resource:
+
+```ruby
+template '/etc/foo' do
+  source 'foo.erb'
+  notifies :run, 'notify_group[Aggregate notifications using a single notify_group resource]', :immediately
+end
+
+template '/etc/bar' do
+  source 'bar.erb'
+  notifies :run, 'notify_group[Aggregate notifications using a single notify_group resource]', :immediately
+end
+
+notify_group 'Aggregate notifications using a single notify_group resource' do
+  notifies :restart, 'service[foo]', :delayed
+end
+```
+
+The `ChefDeprecations/LogResourceNotifications` cop in Cookstyle 6.0 and later detects using the log resource for notifications in cookbooks.
+
+To restore the previous behavior, set `count_log_resource_updates true` in your `client.rb`.
+
+### HWRP Style Resources Now Require resource_name / provides
+
+Legacy HWRP style resources, those written as Ruby classes in the libraries directory of a cookbook, will now require either the `resource_name` or `provides` methods to be used to define the resource names. Previously, Chef Infra Client would infer the desired resource name from the class, but this magic was problematic and has been removed.
+
+The `ChefDeprecations/ResourceWithoutNameOrProvides` cop in Cookstyle 6.0 and later detects this deprecation.
+
+### build_essential GCC Updated on Solaris
+
+On Solaris systems we no longer constrain the version of gcc to 4.8.2 in the build_essential resource to allow for gcc 5 installations.
```

Comment:
Is 'gcc' supposed to be capitalized here? I'm asking since the header on line 57 seems to indicate that for both mentions of 'gcc' in this.

Reply:
It's capitalized now